src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -305,29 +305,43 @@ func KafkaAuthConfig(
 	user, password, mechanism *string,
 ) func() (*config.SCRAM, error) {
 	return func() (*config.SCRAM, error) {
-		if *user == "" && *password == "" {
+		u := *user
+		p := *password
+		m := *mechanism
+		// If the values are empty, check for env vars.
+		if u == "" {
+			u = os.Getenv("REDPANDA_SASL_USERNAME")
```

Comment:
Why not using `viper`? Isn't the whole package `config` under `pkg/config` responsible for handling that?

Reply:
There's a bug in viper's env var + flags bindings. Here's a PR for more context: https://github.com/vectorizedio/redpanda/pull/746

Reply:
ok